leshan-core/src/main/java/org/eclipse/leshan/core/attributes/AttributeSet.java

```diff
@@ -0,0 +1,143 @@
+/*******************************************************************************
+ * Copyright (c) 2013-2018 Sierra Wireless and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *     Sierra Wireless - initial API and implementation
+ *     Daniel Persson (Husqvarna Group) - Attribute support
+ *******************************************************************************/
+package org.eclipse.leshan.core.attributes;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.leshan.util.Validate;
+
+/**
+ * A collection of {@link Attribute} instances that are handled as a collection
+ * that must adhere to rules that are specified in LwM2m, e.g. that the 'pmin' attribute
+ * must be less than the 'pmax' attribute, if they're both part of the same AttributeSet.
+ */
+public class AttributeSet {
+    
+    private Map<AttributeName, Attribute> attributeMap = new LinkedHashMap<>();
```

Comment:
It seems we only need to query by `CoRELinkParam` so I will go for a map String->Attribute